sklearn/compose/_target.py

```diff
@@ -105,7 +106,8 @@ class TransformedTargetRegressor(RegressorMixin, BaseEstimator):
     <sphx_glr_auto_examples_compose_plot_transformed_target.py>`.
 
     """
-    def __init__(self, regressor=None, transformer=None,
+    @_deprecate_positional_args
+    def __init__(self, *, regressor=None, transformer=None,
```

Comment:
Hmm. I think I'm okay with the regressor being specified positionally.

Reply:
First I thought so too, but then I realized there's a `regressor` and a `transformer`, and they kinda are of the same nature, and it'd be easy to confuse which one comes first. That's why I put it before `regressor.

Reply:
hm I would also think after regressor is better. ``transformer`` is optional and I have actually never used it. The main application of this for me is the log transform and we don't have a LogTransformer right now.